api_v3/services/ResponseProfileService.php

```diff
@@ -188,4 +188,30 @@ function recalculateAction(KalturaResponseProfileCacheRecalculateOptions $option
 	{
 		return KalturaResponseProfileCacher::recalculateCacheBySessionType($options);
 	}
+	
+	/**
+	 * List response profiles by filter and pager
+	 * 
+	 * @action clone
+	 * @param int $id
+	 * @throws KalturaErrors::RESPONSE_PROFILE_ID_NOT_FOUND
+	 * @return KalturaResponseProfile
+	 */
+	function cloneAction ($id)
+	{
+		$origResponseProfileDbObject = ResponseProfilePeer::retrieveByPK($id);
+		if (!$origResponseProfileDbObject)
+			throw new KalturaAPIException(KalturaErrors::RESPONSE_PROFILE_ID_NOT_FOUND, $id);
+			
+		$newResponseProfileDbObject = $origResponseProfileDbObject->copy();
+		$newResponseProfileDbObject->setSystemName($origResponseProfileDbObject->getSystemName() . '_' . uniqid());
```

Comment:
Instead of generating system-name, add KalturaResponseProfile as second argument to the action (same as playlist.clone and eventNotificationTemplate.clone) and enforce system-name to be non-null.